src/libmongoc/src/mongoc/mongoc-client.c

```diff
@@ -3126,6 +3133,14 @@ mongoc_client_set_server_api (mongoc_client_t *client,
    return true;
 }
 
+/* Returns true if a specific server API has been set: */
+bool mongoc_client_uses_server_api(mongoc_client_t *client)
+{
+ BSON_ASSERT_PARAM(client);
+
+ return NULL != client->api;
```

Comment:
`mongoc_client_set_server_api` sets `client->api` and calls `_mongoc_topology_scanner_set_server_api`. `mongoc_client_set_server_api` only applies to single-threaded clients.

`mongoc_client_pool_set_server_api` sets `pool->api` and calls `_mongoc_topology_scanner_set_server_api`. `mongoc_client_pool_set_server_api` only applies to pooled clients. One `mongoc_topology_t` is shared among all pooled clients.

Consider having a single accessor, like `mongoc_topology_uses_server_api`. It can apply to both pooled and single threaded clients. It can check if the `topology->scanner->api` is set.